### PR TITLE
mlearning: add Darknet (Open Source Neural Networks in C)

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
Darknet is an open source neural network framework written
in C and CUDA. It is fast, easy to install, and supports
CPU and GPU computation.

You Only Look Once (YOLO) is a state-of-the-art,
real-time object detection system

## Impact
ML

## Testing
ML

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: mlearning/darknet/Makefile [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+CSRCS   +=$(SRC)/yolo_layer.c
+CSRCS   +=$(SRC)/iseg_layer.c
+
+
+CFLAGS += -Wno-shadow -Wno-strict-prototypes -Wno-unknown-pragmas
+
+MODULE = $(CONFIG_DARKNET_YOLO)
+
+darknet.zip:
+	$(Q) curl -L https://github.com/pjreddie/darknet/archive/refs/heads/master.zip -o darknet.zip
```
Comment (pkarashchenko) [inline mlearning/darknet/Makefile:78]: Maybe adding git hash id is a good next step improvement to get better control of what is built

**Review by pkarashchenko — APPROVED** [review]

File: mlearning/darknet/Kconfig [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	default n
+	---help---
+		You only look once (YOLO) is a state-of-the-art,
+                real-time object detection system
+
+if DARKNET_YOLO
+
+config DARKNET_YOLO_VER
+	string "DARKNET YOLO version"
+	default master
```
Comment (xiaoxiang781216) [inline mlearning/darknet/Kconfig:17]: change to “master”

File: mlearning/darknet/Makefile (hunk 1 above)
Comment (xiaoxiang781216) [reply mlearning/darknet/Makefile:78]: let's use DARKNET_YOLO_VER here, so user can select different version through defconfig

Comment (jerpelea) [reply mlearning/darknet/Makefile]: @pkarashchenko I prefer to use stable versions or tags but there is none for YOLO

Comment (pkarashchenko) [reply mlearning/darknet/Makefile]: I mean, that `master` will be different from build to build. I think that using hash has better predictability, so then is can be changed to stable tag once it is available

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@jerpelea do you have plan to support TensorFlow Lite?

**Comment by jerpelea** [comment]
@xiaoxiang781216 I am looking into it

**Pull request merged**